src/_pytest/config/__init__.py

```diff
@@ -847,11 +848,10 @@ def parse(self, args, addopts=True):
                 args, self.option, namespace=self.option
             )
             if not args:
-                cwd = os.getcwd()
-                if cwd == self.rootdir:
+                if self.cwd == self.rootdir:
```

Comment:
i just noticed  this one - its "incorrect"

if test-paths is relative to the original cwd, then the result is entirely incorrect

its absolutely necessary to trigger a warning there - fundamentally a testsuite that changes the cwd at import time for bits of the suite already is broken and users should rectify that

Reply:
@RonnyPfannschmidt 
Ah.. so this should make testpaths absolute then probably.

I agree that `os.chdir` during import etc is bad, but it seemed easy enough to work around / fix this.
Will create a follow-up.